### DISCUSSION TITLE
Pivpn + wireguard stops working after a few days

### DISCUSSION THREAD

**Discussion by Tzal3x** [pivpn/pivpn#1857] [Support]
Hello 👋 ! I have a pivpn & wireguard installation on my raspberry pi 5. 

The issue is that fresh installs of it works, but when a few days pass, I am getting an error like `handshake did not complete after 5 seconds` ( this is from the wireguard logs in my mobile). 

To fix it, I need to completely reinstall the pivpn from scratch.

Why could that be? Any cleaner way to do this?

**Comment by orazioedoardo** [comment]
Likely your public IP changes so you need to either update the client with the new endpoint or use DDNS. You don't need to reinstall.

**Reply by Tzal3x** [reply]
Yea that's the first think I thought of. Connected it with DuckDNS but I still can't access it, getting handshake timeouts again, this time fresh install doesn't work either. 💭

**Reply by orazioedoardo** [reply]
Try the troubleshooting: https://docs.pivpn.io/faq/#how-do-i-troubleshoot-connection-issues

**Reply by Tzal3x** [reply]
All checks seem to be fine, my setup is clean. Is it possible that the router has dropped support for such connections somehow?

**Reply by orazioedoardo** [reply]
Do you receive packets via tcpdump, public IPs match? Port forwarding set?
